homework2/fastq_filtrator.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# coding: utf-8
+def gc_filtering(read, gc_bounds=(0, 100)):
+    gc_num = len([GC for GC in read[1] if GC == "G" or GC == "C"])
+    read_len = len(read[1])
+    gc_content = 100 * gc_num / read_len
+    if gc_bounds[0] <= gc_content <= gc_bounds[1]:
+        return "passed"
+    else:
+        return "rejected"
+
+
+def length_filtering(read, length_bounds=(0, 2 ** 32)):
+    read_len = len(read[1])
+    if length_bounds[0] <= read_len <= length_bounds[1]:
+        return "passed"
+    else:
+        return "rejected"
+
+
+def quality_filtering(read, quality_threshold=0):
+    if sum([ord(i) - 33 for i in read[3]]) / len(read[3]) > quality_threshold:
+        return "passed"
+    else:
+        return "rejected"
```

Comment:
Из подобных функций лучше возвращать логические типы, а не строки, это гораздо удобнее. Да и в итоге можно было бы всё записать в одну строку
```suggestion
def quality_filtering(read, quality_threshold=0):
    return sum([ord(i) - 33 for i in read[3]]) / len(read[3]) >= quality_threshold:
```
Но лучше не надо
```suggestion
def quality_filtering(read, quality_threshold=0):
    average_quality = sum([ord(i) - 33 for i in read[3]]) / len(read[3])
    return average_quality >= quality_threshold:
```

Reply:
Границы надо было включать